chips/nrf52/src/chip.rs

```diff
@@ -26,7 +24,133 @@ impl<I: InterruptService> NRF52<I> {
     }
 }
 
-impl<I: InterruptService> kernel::Chip for NRF52<I> {
+/// This struct, when initialized, instantiates all peripheral drivers for the apollo3.
+/// If a board wishes to use only a subset of these peripherals, this
+/// should not be used or imported, and a modified version should be
+/// constructed manually in main.rs.
+pub struct Nrf52DefaultPeripherals<'a> {
+    pub acomp: crate::acomp::Comparator<'a>,
+    pub ecb: crate::aes::AesECB<'a>,
+    pub gpio_port: &'static crate::gpio::Port<'static>,
```

Comment:
Does this have to be `'static` too?

Reply:
see my other comment on why `gpio_port` did not move to the new model